src/dfx/src/commands/canister/call.rs

```diff
@@ -73,12 +73,17 @@ struct CallIn {
     method_name: String,
     #[serde(with = "serde_bytes")]
     args: Vec<u8>,
-    cycles: u64,
+    cycles: u128,
 }
 
-async fn do_wallet_call(wallet: &Canister<'_, Wallet>, args: &CallIn) -> DfxResult<Vec<u8>> {
-    let (result,): (Result<CallResult, String>,) = wallet
-        .update_("wallet_call")
+async fn do_wallet_call(wallet: &WalletCanister<'_>, args: &CallIn) -> DfxResult<Vec<u8>> {
+    // todo change to wallet.call when IDLValue implements ArgumentDecoder
+    let builder = if wallet.version_supports_u128_cycles() {
```

Comment:
Would you please add some e2e tests that exercise the new behavior?  There are examples in https://github.com/dfinity/sdk/blob/master/e2e/tests-dfx/wallet.bash that test using different wallet versions.

For example, for this piece, tests that show:
- succeeds calling previous wallet version with cycles <64 bits
- fails (reports error) calling previous wallet version, with cycles >64 bits
- succeeds calling new wallet version with cycles >64 bits